src/content/developers/docs/smart-contracts/composability/index.md

```diff
@@ -8,7 +8,61 @@ incomplete: true
+
+A good way to understand composability is to think of composable elements as Lego blocks. Each Lego is designed to fit into another, such that you can build complex structures by combining different Lego blocks. 
+
+In Ethereum, every smart contract is a Lego of sorts—you can use smart contracts from other projects as building blocks for your project. This means you don't have to spend time reinventing the wheel or building from scratch.
+
+## How does composability work? {#how-does-composability-work}
+
+As we said in the beginning, Ethereum smart contracts are like public APIs. This means anyone can interact with them or integrate them into dapps for added functionality.
+
+Smart contract composability generally works off three principles: modularity, autonomy, and discoverability: 
+
+**1. Modularity**: This is the ability of individual components to perform a specific task. In Ethereum, every smart contract has specific use-case (as shown in the Uniswap example).
+
+**2. Autonomy**: Composable components must be able to operate independently. Each smart contract in Ethereum is self-executing and can function without relying on other parts of the system. 
+
+**3. Discoverability**: Developers cannot call external contracts or integrate software libraries into applications if the former are not publicly available. By design, smart contracts are open-source; anyone can call a smart contract or fork a codebase. 
+
+## Benefits of composability {#benefits-of-composability}
+
+### Shorter development cycle
+
+Composability reduces the work that developers have to do when creating [decentralized applications](https://ethereum.org/en/dapps/#what-are-dapps) (dapps). As Naval Ravikant [puts it](https://twitter.com/naval/status/1444366754650656770): "Open source means every problem has to be solved once."
+
+Once a particular smart contract solves one problem, other developers can reuse it, so they don’t have to solve the same problem. In this way, developers can take existing software libraries and add extra functionality to create new dapps. 
+
+### Greater innovation
+
+Composability encourages innovation and experimentation. That’s because developers are free to reuse, modify, duplicate, or integrate open-source code to create desired results. 
+
+Also, development teams spend less time on basic functionality and can allocate more time to experimenting with new features. This explains why there are so many great projects in the Ethereum ecosystem today—developers can focus on improve on existing dapps and build better products. 
+
+### Better user experience
+
+Interoperability between components of the Ethereum ecosystem improves the user experience. Users can access greater functionality when dapps integrate external smart contracts than in a fragmented ecosystem where applications cannot communicate.
```

Comment:
This feels very vague to me. Better user experience because of more feature-rich dapps? I'd (weakly) disagree with this point.

Reply:
What I had in mind was how interactions between different smart contracts increases the functionality available to users. I think flash loans involving multiple dapps are a good example of this (AMMs also fit into this category). It'd be difficult to have a highly interoperable system like that if everyone walled off their apps.

Reply:
Yeah, I see what you're getting at here @emmanuel-awosika , but I think it may help to clarify with at least one example.

Reply:
Hi @wackerow. Thanks for the feedback. I added the flash loan example to explain the point. Let me know if that works.